grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -1879,32 +1879,27 @@ class InventoryService implements ApplicationContextAware {
      * @param transaction
      * @return
      */
-    Boolean isValidForLocalTransfer(Transaction transaction) {
+    void validateForLocalTransfer(Transaction transaction) {
         // make sure that the transaction is of a valid type
         if (transaction?.transactionType?.id != Constants.TRANSFER_IN_TRANSACTION_TYPE_ID &&
                 transaction?.transactionType?.id != Constants.TRANSFER_OUT_TRANSACTION_TYPE_ID) {
-            return false
+            transaction.errors.rejectValue("transactionType", "transaction.localTransfer.invalidType", "Transaction have invalid type for local transfer")
         }
 
         // make sure we are operating only on locally managed warehouses
         if (transaction?.source) {
-            if (!(transaction?.source instanceof Location)) {
+            if (!(transaction?.source instanceof Location) || !transaction?.source?.managedLocally) {
```

Comment:
What weird logic. I don't remember why we had to do this. What else could source have been? :confused:

Reply:
@jmiranda I'm not sure, do you think I should remove this?

Reply:
Added to https://trello.com/b/7uBWaaYi/sprint-75-wonderful-hound